public/app/screens/main/sections/devices/device-detail/device-detail.jsx

```diff
@@ -0,0 +1,106 @@
+var React = require('react');
+var Router = require('react-router');
+var Link = Router.Link;
+var styleMixin = require('mixins/style-mixin');
+var roomStore = require('stores/room');
+var DeviceState = require('./../device-state');
+
+var DeviceDetail = React.createClass({
+  mixins: [
+    styleMixin(require('./style.scss')),
+    Router.State],
+
+
+  getInitialState: function getInitialState () {
+    var device = roomStore.getDevice(this.getParams().deviceId);
+    return {
+      device: device,
+      rooms: roomStore.getRooms(),
+      selectedRoom: roomStore.getRoomFromDevice(device)
+    };
+  },
+
+  getRoomDropDown: function getRoomDropDown () {
```

Comment:
I initially didn't see it. This method only returns the `option` tags for `select`. Thus this method should be called `getRoomSelectOptions` or `getRoomDropDownOptions`. I prefer the first.